src/acvp_aes.c

```diff
@@ -1042,19 +1067,43 @@ ACVP_RESULT acvp_aes_kat_handler(ACVP_CTX *ctx, JSON_Object *obj) {
                                         && dir == ACVP_SYM_CIPH_DIR_ENCRYPT &&
                                         iv_gen == ACVP_SYM_CIPH_IVGEN_SRC_INT)) {
                 if (alg_id == ACVP_AES_XTS) {
-                    /* XTS may call it tweak value, but we treat it as an IV */
-                    iv = json_object_get_string(testobj, "tweakValue");
-                    if (!iv) {
-                        ACVP_LOG_ERR("Server JSON missing 'tweakValue'");
+                    tweak_mode = read_tw_mode(tw_mode);
+                    if (!tweak_mode) {
+                        ACVP_LOG_ERR("Server JSON wrong 'tweakMode'");
                         rv = ACVP_MISSING_ARG;
                         goto err;
                     }
-                    if (strnlen_s(iv, ACVP_SYM_IV_MAX + 1) > ACVP_SYM_IV_MAX) {
-                        ACVP_LOG_ERR("'i' too long, max allowed=(%d)",
-                                     ACVP_SYM_IV_MAX);
-                        rv = ACVP_INVALID_ARG;
-                        goto err;
-                    }
+                    switch (tweak_mode) {
+                        case ACVP_SYM_CIPH_TWEAK_HEX:
+                            /* XTS may call it tweak value, but we treat it as an IV */
+                            iv = json_object_get_string(testobj, "tweakValue");
+                            if (!iv) {
+                                ACVP_LOG_ERR("Server JSON missing hex 'tweakValue'");
+                                rv = ACVP_MISSING_ARG;
+                                goto err;
+                            }
+                            if (strnlen_s(iv, ACVP_SYM_IV_MAX + 1) > ACVP_SYM_IV_MAX) {
+                                ACVP_LOG_ERR("'i' too long, max allowed=(%d)",
+                                             ACVP_SYM_IV_MAX);
+                                rv = ACVP_INVALID_ARG;
+                                goto err;
+                            }
+                            break;
+                        case ACVP_SYM_CIPH_TWEAK_NUM:
+                            seq_num = json_object_get_number(testobj, "sequenceNumber");
+                            if ((seq_num < 0) || (seq_num > 255)) {
+                                ACVP_LOG_ERR("Server JSON invalid number 'tweakValue'");
+                                rv = ACVP_MISSING_ARG;
+                                goto err;
+                            }
+                            break;
+                        case ACVP_SYM_CIPH_TWEAK_NONE:
+                        default:
+                            ACVP_LOG_ERR("Server JSON invalid 'tweakMode'");
+                            rv = ACVP_MISSING_ARG;
+                            goto err;
```

Comment:
Coverity possible complaint for lack of break

Reply:
fixed